### PR TITLE
Fix broken Django Compressor setup and enable static files minification

### LINKED ISSUE #2336: Broken setup with use_whitenoise=y, use_compressor=y and cloud_provider=none
## What happened?
Multiple things:
- Altough, we set cloud_provider to None, follwing is set in `config/settings/production.py`:
`COMPRESS_STORAGE = "storages.backends.s3boto3.S3Boto3Storage"`
- Compressed static files are not available via Whitenoise, giving an error on page:

`compressor.exceptions.OfflineGenerationError: You have offline compression enabled but key "..." is missing from offline manifest. You may need to run "python manage.py compress"`

## What should've happened instead?
- Compress storage should be set to local configuration:
`COMPRESS_STORAGE = "compressor.storage.GzipCompressorFileStorage"`
- `compose/production/django/start` should be extended to this:
```
python /app/manage.py collectstatic --noinput
python /app/manage.py compress --force
python /app/manage.py collectstatic --noinput
```

See https://stackoverflow.com/questions/58712195/django-whitenoise-with-compressed-staticfiles for further information

## Steps to reproduce
Set up project with following settings:
- use_docker=y
- use_whitenoise=y
- use_compressor=y
- cloud_provider=3 (None)

### PR DESCRIPTION
**Pull request by jameswilliams1**
[//]: # (Thank you for helping us out: your efforts mean great deal to the project and the community as a whole!)

[//]: # (Before you proceed:)

[//]: # (1. Make sure to add yourself to `CONTRIBUTORS.rst` through this PR provided you're contributing here for the first time)
[//]: # (2. Don't forget to update the `docs/` presuming others would benefit from a concise description of whatever that you're proposing)

## Description

[//]: # (What's it you're proposing?)
To change the django compressor setup as well as the startup script for django when required (if using Whitenoise).

## Rationale

[//]: # (Why does the project need that?)
I am propsing to change the django compressor setup for 2 reasons:
1) It currently does not work except when chosing S3 cloud storage
2) Currently no minification is performed on files, making it mostly redundant as this is a major feature of compressor.

The start script is modified when whitenoise is also enabled as compress must then run at start (part of the reason it currently doesn't work).

## Use case(s) / visualization(s)

[//]: # ("Better to see something once than to hear about it a thousand times.")
It will improve site speed by minifiying static resources on app start and also fixes a broken setup.

Fixes #2336

### CODE REVIEW

**Comment by jameswilliams1** [comment]
Fixes issue #2336

File: {{cookiecutter.project_slug}}/compose/production/django/start [hunk 1]
```diff
@@ -6,4 +6,10 @@ set -o nounset
 
 
 python /app/manage.py collectstatic --noinput
+{%- if cookiecutter.use_whitenoise == 'y' and cookiecutter.use_compressor == 'y' %}
+if [ "${COMPRESS_ENABLED:-}" = "True" ];
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/compose/production/django/start]: I'm not sure about that... As far as I can tell, this is unset in prod, and the setting default to `True` if absent. This bash `if` will never evaluate run, will it? Shouldn't it be that instead:

```suggestion
if [ "${COMPRESS_ENABLED:-True}" = "True" ];
```

Also, the way django-environ reads the environment, it [supports several values as truthy](https://django-environ.readthedocs.io/en/latest/index.html#environ.environ.Env.BOOLEAN_TRUE_STRINGS), but this supports only one.

I think I'd be OK to keep it simple with just the template branch from line 9, but I'm open to a better idea.

Comment (jameswilliams1) [reply {{cookiecutter.project_slug}}/compose/production/django/start]: Of course, I hadn't thought of that as I'm using this as an env var on my site. The issue is if compress is disabled and this line runs it causes an error, so without some check here there's no way of disabling compressor using env vars (this file would also have to change). I think the default to true should mostly work for a lot of cases though, I will change this.

Comment (jameswilliams1) [reply {{cookiecutter.project_slug}}/compose/production/django/start]: @browniebroke I think this solution using environ should cover any edge cases now. Broken link also fixed.

**Comment by browniebroke** [comment]
I gave this a try, but it's not working out of the box. If I choose to use compressor, the compression doesn't run by default. I think we're just missing one piece, it's to change the default environment variable here, which is unset:

https://github.com/pydanny/cookiecutter-django/blob/530868d072248b4472fb09a42e511c6024f361dd/%7B%7Bcookiecutter.project_slug%7D%7D/.envs/.production/.django#L34-L38

Since the default value is [already specified in the settings](https://github.com/pydanny/cookiecutter-django/blob/530868d072248b4472fb09a42e511c6024f361dd/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L202), and this is not anything secret, I suggest to remove the above part from the `.envs/.production/.django` entirely. Thoughts?

**Comment by jameswilliams1** [comment]
> 
> 
> I gave this a try, but it's not working out of the box. If I choose to use compressor, the compression doesn't run by default. I think we're just missing one piece, it's to change the default environment variable here, which is unset:
> 
> https://github.com/pydanny/cookiecutter-django/blob/530868d072248b4472fb09a42e511c6024f361dd/%7B%7Bcookiecutter.project_slug%7D%7D/.envs/.production/.django#L34-L38
> 
> Since the default value is [already specified in the settings](https://github.com/pydanny/cookiecutter-django/blob/530868d072248b4472fb09a42e511c6024f361dd/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L202), and this is not anything secret, I suggest to remove the above part from the `.envs/.production/.django` entirely. Thoughts?

Strange, hadn't noticed that part but it makes sense to just remove it from the .django entirely (can't see any reason why someone would want it unset? probably a bug). I'll update tomrrow when at a PC.

**Comment by browniebroke** [comment]
Thanks for the quick reply.

**Comment by browniebroke** [comment]
Thanks 🎉

**Pull request merged**